src/coreclr/inc/clrdata.idl

```diff
@@ -308,9 +308,10 @@ interface ICLRDataLoggingCallback : IUnknown
 typedef enum CLRDataEnumMemoryFlags
 {
     CLRDATA_ENUM_MEM_DEFAULT = 0x0,
-    CLRDATA_ENUM_MEM_MINI = CLRDATA_ENUM_MEM_DEFAULT,       // generating skinny mini-dump
-    CLRDATA_ENUM_MEM_HEAP = 0x1,        // generating heap dump
-    CLRDATA_ENUM_MEM_TRIAGE = 0x2,      // generating triage mini-dump
+    CLRDATA_ENUM_MEM_MINI = CLRDATA_ENUM_MEM_DEFAULT,   // generating skinny mini-dump
+    CLRDATA_ENUM_MEM_HEAP = 0x1,                        // generating heap dump
+    CLRDATA_ENUM_MEM_TRIAGE = 0x2,                      // generating triage mini-dump
+    CLRDATA_ENUM_MEM_HEAP2 = 0x3,                       // generating heap dump faster with less memory usage
```

Comment:
Added a little better comment. As far as non-essential changes the only one I see that is significant is removing the BaseDomain's EnumMemoryRegions since it didn't do anything.